src/main/java/uk/gov/hmcts/reform/et/syaapi/service/CaseDocumentService.java

```diff
@@ -164,17 +153,21 @@ private void validateFile(MultipartFile file) throws CaseDocumentException, IOEx
 
     private CaseDocument validateResponse(DocumentUploadResponse response, String originalFilename)
         throws CaseDocumentException {
-        if (response.documents == null || response.isEmpty()) {
+        if (response.getDocuments() == null) {
             throw new CaseDocumentException(UPLOAD_FILE_EXCEPTION_MESSAGE + originalFilename);
         }
 
         CaseDocument document = response.getDocuments().stream()
             .findFirst()
             .orElseThrow(() -> new CaseDocumentException(UPLOAD_FILE_EXCEPTION_MESSAGE + originalFilename));
 
-        String uri = getUriFromFile(document, originalFilename).toString();
+        if (document.getLinks() == null
+            || document.getLinks().get("self") == null
+            || document.getUri() == null) {
```

Comment:
Validating the "self" link's URL might be better done alongside the getUri method on CaseDocument.  Still verify and throw the exception here, but would add value doing the checks in there